/*
 * Copyright 2016-2019 Johns Hopkins University
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * Developed by: Roberts Group
 *               Johns Hopkins University
 *               http://biophysics.jhu.edu/roberts/
 *
 * Author(s): Elijah Roberts, Max Klein
 */

syntax = "proto2";
package lm.io.ffpilot;

import "robertslab/pbuf/NDArray.proto";

message FFPilotStageOutput {

    required int32 id                                               = 1;

    enum Direction {
        FORWARD                 = 0;
        BACKWARD                = 1;
    }
    required Direction direction                                    = 2;

    enum Type {
        PILOT                   = 0;
        PRODUCTION		= 1;
    }
    required Type type                                              = 3;

    // The locations of the edges in the order parameter coordinate.
    required robertslab.pbuf.NDArray edges                          = 10;    //1D; rows=edges; type=float64

    // The number of trajectories launched in each phase.
    required robertslab.pbuf.NDArray trajectory_counts              = 11;    //1D; rows=edges; type=uint64

    // The average computational cost of a single trajectory run in each phase.
    required robertslab.pbuf.NDArray costs                          = 12;    //1D; rows=edges; type=float64

    // The weight calculated at the end of each phase.
    required robertslab.pbuf.NDArray weights                        = 13;    //1D; rows=edges; type=float64

    // The variance in the phase weights.
    required robertslab.pbuf.NDArray weight_variances               = 15;    //1D; rows=edges; type=float64

    // The number of trajectories launched in each phase.
    optional robertslab.pbuf.NDArray optimized_trajectory_counts    = 16;    //1D; rows=edges; type=uint64

    // First passage times with respect to tiling edge i.
    optional robertslab.pbuf.NDArray first_passage_times            = 14;    //1D; rows=edges; type=float64
}
